code/ship/ship.cpp

```diff
@@ -11234,6 +11233,16 @@ int ship_fire_primary(object * obj, int stream_weapons, int force)
 							// of weapon_create							
 							weapon_objnum = weapon_create( &firing_pos, &firing_orient, weapon_idx, OBJ_INDEX(obj), new_group_id,
 								0, 0, swp->primary_bank_fof_cooldown[bank_to_fire] );
+
+							if (weapon_objnum == -1) {
```

Comment:
This only handles the case when a primary weapon is fired but not secondaries. Does this feature only support primary weapons? I do not see anything in the new parsing code that would enforce this.

Reply:
This was an oversight. Spoon specified this specifically for primary weapons so I overlooked secondary weapons. I will fix that later today.